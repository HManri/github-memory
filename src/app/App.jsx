import React from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';

import MainLayout from 'layout/MainLayout';
import Home from 'pages/Home';
import Match from 'pages/Match';

export default function App() {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<MainLayout />}>
          <Route index element={<Home />} />
          <Route path="match" element={<Match />} />
        </Route>
      </Routes>
    </BrowserRouter>
  );
}
