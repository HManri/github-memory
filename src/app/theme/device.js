const size = {
  mobile: '425px',
  desktop: '768px',
};

export const device = {
  mobile: `(min-width: ${size.mobile})`,
  desktop: `(min-width: ${size.desktop})`,
};
