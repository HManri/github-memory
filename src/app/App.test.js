import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import theme from 'theme/defaultTheme';
import App from 'App.jsx';

describe('App', () => {
  test('should render correctly with Router', () => {
    const app = render(
      <ThemeProvider theme={theme}>
        <App />
      </ThemeProvider>,
    );

    expect(app).toMatchSnapshot();
  });
});
