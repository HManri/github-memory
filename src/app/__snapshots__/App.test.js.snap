// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly with Router 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bcXHqe lbAtZn"
        data-testid="app__banner"
      >
        <div
          class="sc-gswNZR fpcRz"
        >
          GitHub Memory
        </div>
      </div>
      <div
        class="sc-dkrFOg dmtuQU"
        data-testid="app__wrapper"
      >
        <div
          class="sc-eDvSVe ewnMrV"
          data-testid="home"
        >
          <div
            class="sc-jSUZER ghjMJT"
            data-testid="home__box"
          >
            <div
              class="sc-gKPRtg fVBoFd"
              data-testid="home__title"
            >
              <div
                class="sc-iBYQkv fASOKr"
              >
                Welcome to
              </div>
              <div
                class="sc-ftTHYK hTSZkf"
              >
                GitHub Memory
              </div>
            </div>
            <div
              class="sc-jrcTuL"
              data-testid="home__actions"
            >
              <button
                class="sc-hLBbgP hIGdbd"
                data-testid="start_game__btn"
              >
                <span
                  class="sc-pyfCe kdlpgc"
                >
                  Start
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bcXHqe lbAtZn"
      data-testid="app__banner"
    >
      <div
        class="sc-gswNZR fpcRz"
      >
        GitHub Memory
      </div>
    </div>
    <div
      class="sc-dkrFOg dmtuQU"
      data-testid="app__wrapper"
    >
      <div
        class="sc-eDvSVe ewnMrV"
        data-testid="home"
      >
        <div
          class="sc-jSUZER ghjMJT"
          data-testid="home__box"
        >
          <div
            class="sc-gKPRtg fVBoFd"
            data-testid="home__title"
          >
            <div
              class="sc-iBYQkv fASOKr"
            >
              Welcome to
            </div>
            <div
              class="sc-ftTHYK hTSZkf"
            >
              GitHub Memory
            </div>
          </div>
          <div
            class="sc-jrcTuL"
            data-testid="home__actions"
          >
            <button
              class="sc-hLBbgP hIGdbd"
              data-testid="start_game__btn"
            >
              <span
                class="sc-pyfCe kdlpgc"
              >
                Start
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
